nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py

```diff
@@ -399,6 +402,106 @@ def _get_text(self, idx: int) -> np.ndarray:
             sample = np.pad(
                 sample, (0, self.seq_length + self.add_extra_token - len(sample)), mode='constant', constant_values=-1
             )
+
+        if not self.cfg.data.get('add_fim', False):
+            return sample.astype(np.int64)
+        # FIM implementation
+        else:
+            # get FIM params
+            self.fim_rate = self.cfg.data.fim.rate
+            self.fim_spm_rate = self.cfg.data.fim.spm_rate
+            self.fim_split_sample = self.tokenizer.tokens_to_ids(self.cfg.data.fim.split_sample)
```

Comment:
@dimapihtar @suiyoubi  does `tokenizer.tokens_to_ids` return a list of token-ids?
If yes, then this is an issue, because this code https://github.com/NVIDIA/NeMo/blob/ca356e870037a63bf97634fa4c5efab7c43490bc/nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py#L458 expects it to be a single token-id, not a list.
Then the sample-splitting around this token would never occur. This would explain the low FIM-results we are observing